xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEBuffer.cpp

```diff
@@ -233,7 +233,7 @@ bool CActiveAEBufferPoolResample::Create(unsigned int totaltime, bool remap, boo
       if (m_processor->GetChannelLayout().Count() > 2)                    /* Disable upmix for CActiveAEResample if DSP layout > 2.0, becomes perfomed by DSP */
         upmix = false;
 
-      m_dspBuffer = new CActiveAEBufferPool(m_inputFormat);               /* Get dsp processing buffer class, based on dsp output format */
+      m_dspBuffer = new CActiveAEBufferPool(m_format);               /* Get dsp processing buffer class, based on dsp output format */
```

Comment:
@FernetMenta or probably @Memphiz 
Do you have the time so send me an PR for XCode?  I moved the entire source code of `AudioDSP from xbmc/cores/AudioEngine` to `xbmc/cores/AudioEngine/Engines/ActiveAE/AudioDSPAddons/`.